graylog2-web-interface/src/components/rules/RuleForm.test.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { render } from 'wrappedTestingLibrary';
+
+import RuleForm from './RuleForm';
+import { PipelineRulesProvider } from './RuleContext';
+
+describe('RuleForm', () => {
+  it('should show the previous Description value in the field', () => {
```

Comment:
```suggestion
  it('should show the previous description value in the field', () => {
```

Reply:
Maybe I am not understanding the test correctly but I think it does not test the bug fixed in the reference ticket. If I got it right the bug was that the description was not saved correctly caused by the field not updated properly.

I'd change this test:
- remove the description to be rendered by default
- use a user event in the input field to change the description
- mock `handleSavePipelineRule` (part of `PipelineRulesContext`) and check that it was called with the correct attributes

This test would be closer to the user behaviour and is thus (a bit) more independent from the implementation details.

WDYT?

Reply:
Basically the problem was that the field was always empty, I agree with your suggestion to make the test as closer to user behaviour as possible it is better to mock the `handleDescription` and make sure it is always called with the correct `description` value.